### ISSUE TITLE
upload files to cloudinary

### ISSUE THREAD

**Issue by leviwheatcroft** [superwolff/metalsmith-cloudinary#5]
Hey I know you have superwolff/cloudinate but I want something to upload files from the metalsmith files structure. I'd like to implement this myself but I thought I'd run it by you before I start on a big PR like this.

I did take a look at cloudinate but it's mostly cli boilerplate, almost the entire codebase would need to be rewritten even if you wanted to make a metalsmith-cloudinate plugin which leverages the cloudinate repo.

My use case is that I'm populating the metalsmith files structure with metalsmith-google-drive, so the idea is that my build process could both download images from google drive and then upload them to cloudinary, without storing them on the local harddrive.

I know there's this "do one thing well" paradigm with metalsmith plugins but really, is two separate operations on the same api "one thing" or two ?

If you did want to include this kind of push functionality in this plugin, I think the best way to implement would be by simply adding a `push`: `minimatch pattern` property to the plugin options. So calling the plugin would always read the available image meta to the `cloudinary` meta property (like the existing functionality), but if you specify a `push` pattern then the additional upload functionality is applied.

**Comment by ismay** [comment]
That's a good idea, I personally don't think it would be too much for a single plugin. As long as the code stays manageable it should be fine 👍  I could give you access to this repo if that makes the contribution process easier?

**Comment by leviwheatcroft** [comment]
Yeah access would be great. I'll fix the current open PR, then create a new branch for the push functionality.

**Comment by ismay** [comment]
> Yeah access would be great.

Allright, invite sent.
